cli/command/idresolver/idresolver_test.go

```diff
@@ -0,0 +1,146 @@
+package idresolver
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/cli/internal/test"
+	// Import builders to get the builder function as package function
+	. "github.com/docker/docker/cli/internal/test/builders"
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+func TestResolveError(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		nodeInspectFunc: func(nodeID string) (swarm.Node, []byte, error) {
+			return swarm.Node{}, []byte{}, errors.Errorf("error inspecting node")
+		},
+	}, &bytes.Buffer{})
+
+	idResolver := New(cli.Client(), false)
+	_, err := idResolver.Resolve(context.Background(), struct{}{}, "nodeID")
+
+	assert.Error(t, err, "unsupported type")
+}
+
+func TestResolveWithNoResolveOption(t *testing.T) {
+	resolved := false
+	cli := test.NewFakeCli(&fakeClient{
```

Comment:
Thank you for the review. I updated the PR to create and directly pass the client.